Allow dev imports to pull from spigot libraries

Machine-Maker opened the pull request:
closes #131

MiniDigger reviewed (approved):
havent tested but this looks sane

jpenilla reviewed (changes requested):
As mentioned internally the filtering should use the libraries txt not the result of another download, to allow the downloads to run in parallel

Machine-Maker commented:
Ok, switched to using the libraries txt file. Also added maven central as a fallback repo when downloading dependencies and sources for the vanilla libraries. Looks like mojang doesn't now (or never did) mirror the source jars, just the library jars.

jpenilla commented on paperweight-lib/src/main/kotlin/tasks/ApplyPaperPatches.kt:
```diff
@@ -123,7 +126,7 @@ abstract class ApplyPaperPatches : ControllableOutputTask() {
                 decompJar = sourceMcDevJar.path,
                 importsFile = devImports.pathOrNull,
-                librariesDir = mcLibrariesDir.path,
+                librariesDirs = listOf(mcLibrariesDir.path, spigotLibrariesDir.path),
```
we want spigot first here in case they override a dep version

jpenilla commented on paperweight-patcher/src/main/kotlin/tasks/SimpleApplyGitPatches.kt:
```diff
@@ -119,13 +124,16 @@ abstract class SimpleApplyGitPatches : ControllableOutputTask() {
 
         val patches = patchDir.pathOrNull?.listDirectoryEntries("*.patch") ?: listOf()
+        val librarySources = ArrayList<Path>()
+        mcLibrariesDir.pathOrNull?.let { librarySources.add(it) }
```
again spigot sources first

jpenilla commented on paperweight-lib/src/main/kotlin/util/McDev.kt:
```diff
@@ -41,7 +41,7 @@ object McDev {
         decompJar: Path,
         importsFile: Path?,
-        librariesDir: Path?,
+        librariesDirs: List<Path>?,
```
move this argument down one line, make it non null, and default value of `listOf()`
```kotlin
librariesDirs: List<Path> = listOf(),
```

callers will need updating ofc
